terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -106,17 +119,23 @@ def source_from(details_hash)
           when :github, :bitbucket, :git
             git_source_details_from(bare_source)
           when :registry
-            registry_source_details_from(bare_source)
+            registry_source_details_from(bare_source, provider)
           end
 
         source_details[:proxy_url] = raw_source if raw_source != bare_source
         source_details
       end
 
-      def registry_source_details_from(source_string)
+      def registry_source_details_from(source_string, provider)
         parts = source_string.split("//").first.split("/")
 
-        if parts.count == 3
+        if provider && parts.count == 2
```

Comment:
I'm not happy having to pass down this 'provider' boolean this deeply, but I didn't find a clean way yet without having a bunch of duplication 🤔 

Would love suggestions on how to make this suck less

Reply:
🤔 I took a peek at https://www.terraform.io/docs/language/providers/requirements.html#source-addresses which says:

> `[<HOSTNAME>/]<NAMESPACE>/<TYPE>`

Here's an idea:

```ruby
host, namespace, type = source_address_from(source_string)

def source_address_from(source_string)
  parts = source_string.split('/')
  parts.length == 2 ? ['registry.terraform.io', parts[0], parts[1]] : parts
end
```

Feel free to ignore.

Reply:
Yeah I had considered a similar approach, but the downside is that if someone has an invalid module configured, we would end up trying to treat it as a provider. We *know* if we're dealing with a module or a provider, so trying to assume it seems slightly off? 🤔

Reply:
> Yeah I had considered a similar approach, but the downside is that if someone has an invalid module configured, we would end up trying to treat it as a provider. We _know_ if we're dealing with a module or a provider, so trying to assume it seems slightly off? 🤔

I don't quite see where modules fit into this. Is it possible that this comment was meant for [this suggestion](https://github.com/dependabot/dependabot-core/pull/3729#discussion_r634666201) or is there more that I'm not seeing yet.